paimon-core/src/main/java/org/apache/paimon/metrics/AbstractMetricGroup.java

```diff
@@ -0,0 +1,201 @@
+ * group is adding metrics to the map at the same time.
+ */
+public abstract class AbstractMetricGroup implements MetricGroup {
+    protected static final Logger LOG = LoggerFactory.getLogger(MetricGroup.class);
+
+    // ------------------------------------------------------------------------
+
+    /** The map containing all tags and their associated values, lazily computed. */
+    protected Map<String, String> tags;
+
+    /** Flag indicating whether this group has been closed. */
+    private boolean closed = false;
+
+    private final ConcurrentHashMap<String, Metric> metrics = new ConcurrentHashMap<>();
+
+    // ------------------------------------------------------------------------
+
+    public AbstractMetricGroup(@Nullable Map<String, String> tags) {
+        this.tags = tags;
+        Metrics.getInstance().addGroup(this);
+    }
+
+    @Override
+    public Map<String, String> getAllTags() {
+        return tags;
+    }
+
+    /**
+     * Returns the fully qualified metric name using the configured delimiter for the reporter with
+     * the given index, for example {@code "myTable.bucket-1.metricName"}.
+     *
+     * @param metricName metric name
+     * @param delimiter delimiter to use
+     * @return fully qualified metric name
+     */
+    public String getMetricIdentifier(String metricName, String delimiter) {
+        return String.join(delimiter, getGroupName(), metricName);
+    }
+
+    /**
+     * Creates and registers a new {@link org.apache.paimon.metrics.Counter} or return the existing
+     * {@link org.apache.paimon.metrics.Counter}.
+     *
+     * @param name name of the counter
+     * @return the created or existing counter
+     */
+    public Counter counter(String name) {
+        return counter(name, new SimpleCounter());
+    }
+
+    /**
+     * Registers a {@link org.apache.paimon.metrics.Counter}.
+     *
+     * @param name name of the counter
+     * @param counter counter to register
+     * @param <C> counter type
+     * @return the given counter
+     */
+    public <C extends Counter> C counter(String name, C counter) {
+        Metric metric = addMetric(name, counter);
+        return (C) metric;
+    }
+
+    /**
+     * Registers a new {@link org.apache.paimon.metrics.Gauge}.
+     *
+     * @param name name of the gauge
+     * @param gauge gauge to register
+     * @param <T> return type of the gauge
+     * @return the given gauge
+     */
+    public <T, G extends Gauge<T>> G gauge(String name, G gauge) {
+        Metric metric = addMetric(name, gauge);
+        return (G) metric;
+    }
+
+    /**
+     * Registers a new {@link Histogram} with Paimon.
+     *
+     * @param name name of the histogram
+     * @param histogram histogram to register
+     * @param <H> histogram type
+     * @return the registered histogram
+     */
+    public <H extends Histogram> H histogram(String name, H histogram) {
+        Metric metric = addMetric(name, histogram);
+        return (H) metric;
```

Comment:
```suggestion
        return (H) addMetric(name, histogram);
```

Reply:
Yes, this is a redundant coding, updated it.